pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go

```diff
@@ -68,9 +68,15 @@ func (db *AzureSqlDbManager) Ensure(ctx context.Context, obj runtime.Object, opt
 
 	// Before we attempt to issue a new update, check if there is a previously ongoing update
 	if instance.Status.PollingURL != "" {
+		// TODO: There are other places which use PollClient which may or may not need this treatment as well...
 		pClient := pollclient.NewPollClient()
 		res, err := pClient.Get(ctx, instance.Status.PollingURL)
-		if err != nil {
+		pollErr := errhelp.NewAzureErrorAzureError(err)
```

Comment:
Can Get ever return a non-AzureError error error?

Reply:
The way `NewAzureErrorAzureError` is written, even if the error doesn't match one of the predefined types it does its best to match what it can, so it won't blow up.

I don't think that get will return anything other than `autorest.DetailedError` (which is what `NewAzureErrorAzureError` parses), and there are other places in the code where I see `NewAzureErrorAzureError` has been used on the result of a call to the polling client... so I think we're ok here.

Reply:
That makes sense - thanks for the extra detail!